app/src/main/java/com/github/libretube/util/TextUtils.kt

```diff
@@ -66,19 +52,12 @@ object TextUtils {
      * Get video id if the link is a valid youtube video link
      */
     fun getVideoIdFromUri(link: String): String? {
-        val uri = Uri.parse(link)
-
-        if (link.contains("youtube.com")) {
-            // the link may be in an unsupported format, so we should try/catch it
-            return try {
-                uri.getQueryParameter("v")
-            } catch (e: Exception) {
-                null
+        return link.toHttpUrlOrNull()?.let {
+            when (it.host) {
+                "youtube.com" -> it.queryParameter("v")
```

Comment:
Couldn't this fail when the host starts with `www.`?

Reply:
Yeah, my bad. Updated.